app/src/main/java/com/example/harvestfresh/SignupActivity.java

```diff
@@ -0,0 +1,75 @@
+package com.example.harvestfresh;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.parse.ParseException;
+import com.parse.ParseUser;
+import com.parse.SignUpCallback;
+
+public class SignupActivity extends AppCompatActivity {
+
+    private Button btnFinalSignup;
+    private EditText etSignupUsername;
+    private EditText etSignupPassword;
+    private EditText etConfirmPassword;
+    private static final String TAG = "SIGNUPACTIVITY";
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.signup_activity);
+
+        btnFinalSignup = findViewById(R.id.btnFinalSignup);
+        etSignupUsername = findViewById(R.id.etSignupUsername);
+        etSignupPassword = findViewById(R.id.etSignupPassword);
+        etConfirmPassword = findViewById(R.id.etConfirmPassword);
+
+
+        btnFinalSignup.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String newUsername = etSignupUsername.getText().toString();
+                String newPassword = etSignupPassword.getText().toString();
+                String confirmPassword = etConfirmPassword.getText().toString();
+
+                if(newUsername.isEmpty() || newPassword.isEmpty() || confirmPassword.isEmpty()){
```

Comment:
Nit: For long, repetitive lists of code like this, it looks a bit cleaner to put it on multiple lines:
if(newUsername.isEmpty() || 
   newPassword.isEmpty() || 
   confirmPassword.isEmpty()) {
   
   
   Also, I notice that sometimes you're putting a space before the open bracket, like on line 17 36 here, and sometimes, you don't, such as here on 43 or 46 below. Pick one convention and stick with it (I prefer space before bracket personally)